File: ecommerce-api-legacy/src/utils/logger.js
// Logger estruturado e centralizado (RP-08 corrige AP-11).
// Nunca logar segredos, chaves de gateway ou dados de cartão (PAN).
function emit(level, message, meta) {
    const base = `[${new Date().toISOString()}] ${level.toUpperCase()} ${message}`;
    const line = meta ? `${base} ${JSON.stringify(meta)}` : base;
    (level === 'error' ? console.error : console.log)(line);
}

module.exports = {
    error: (message, meta) => emit('error', message, meta),
    warn: (message, meta) => emit('warn', message, meta),
    info: (message, meta) => emit('info', message, meta),
    debug: (message, meta) => emit('debug', message, meta),
};

File: task-manager-api/utils/logger.py
import logging

# Logging estruturado e centralizado (RP-08 corrige AP-11).
logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s %(levelname)s %(name)s %(message)s',
)


def get_logger(name):
    return logging.getLogger(name)
